internal/services/videoanalyzer/video_analyzer_edge_module_resource.go

```diff
@@ -0,0 +1,136 @@
+
+			"video_analyzer_name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.VideoAnalyzerName(),
+			},
+		},
+	}
+}
+
+func resourceVideoAnalyzerEdgeModuleCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.EdgeModulesClient
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	ctx, cancel := timeouts.ForCreateUpdate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+	resourceId := parse.NewEdgeModulesID(subscriptionId, d.Get("resource_group_name").(string), d.Get("video_analyzer_name").(string), d.Get("name").(string))
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, resourceId.ResourceGroup, resourceId.VideoAnalyzerName, resourceId.EdgeModuleName)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %s: %+v", resourceId, err)
+			}
+		}
+
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_video_analyzer_edge_module", resourceId.ID())
+		}
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, resourceId.ResourceGroup, resourceId.VideoAnalyzerName, resourceId.EdgeModuleName, videoanalyzer.EdgeModuleEntity{}); err != nil {
+		return fmt.Errorf("creating %s: %+v", resourceId, err)
+	}
+
+	d.SetId(resourceId.ID())
+	return resourceVideoAnalyzerEdgeModuleRead(d, meta)
+}
+
+func resourceVideoAnalyzerEdgeModuleRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.EdgeModulesClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.EdgeModulesID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Get(ctx, id.ResourceGroup, id.VideoAnalyzerName, id.EdgeModuleName)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			log.Printf("[INFO] Video Analyzer Edge Module %q was not found in Resource Group %q - removing from state", id.EdgeModuleName, id.ResourceGroup)
+			d.SetId("")
+			return nil
+		}
+
+		return fmt.Errorf("retrieving Video Analyzer Edge Module %q (Resource Group %q): %+v", id.EdgeModuleName, id.ResourceGroup, err)
+	}
+
+	d.Set("name", id.EdgeModuleName)
+	d.Set("resource_group_name", id.ResourceGroup)
+	d.Set("video_analyzer_name", id.VideoAnalyzerName)
+
+	return nil
+}
+
+func resourceVideoAnalyzerEdgeModuleDelete(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).VideoAnalyzer.EdgeModulesClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := parse.EdgeModulesID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resp, err := client.Delete(ctx, id.ResourceGroup, id.VideoAnalyzerName, id.EdgeModuleName)
+	if err != nil {
+		if response.WasNotFound(resp.Response) {
+			return nil
+		}
```

Comment:
can we remove this, this is a bug we're gradually removing from resources (#12879 does this for most of the codebase)

```suggestion
```

Reply:
Ah, darn it, I remembered that and removed it but then forgot to push the change last night 😿